### PR TITLE
Feat/use colors

### PR DESCRIPTION
**Pull request by ljosberinn**
## Pull request checklist

Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been reviewed and added / updated if needed (for bug fixes
      /start features)

## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

E.g. `linear-gradient` as css property needs inline knowledge of resolved colors, this is currently not possible with 
```js
<Box background="linear-gradient(teal.100, red.300)" />
```

Until this is resolved, `useColors` allows to resolve the colors without having to use 

```js
bg={`linear-gradient(${theme.colors.teal[100]}, ${theme.colors.red[300]})`}
```
 through a theme import.

## What is the new behavior?

- `useToken` resolves any design tokens safely from the theme
- removed `memoizeOne`
- implemented caching of anything `get` is used with via a `WeakMap`
- fixed failed develop tests

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by ljosberinn** [comment]
Snuck in [`jest-watch-typeahead`](https://github.com/jest-community/jest-watch-typeahead#readme) which lets you filter testfiles by input and with keyup/down:
image

**Review by with-heart — APPROVED** [review]

**Comment by ljosberinn** [comment]
Talked about it a bit and settled for a more flexible API instead:
```js
function useToken(key, ...tokens) {
  const theme = useTheme();

  return tokens.map(token => theme[key]?.[token] ?? undefined)
}
```

File: packages/utils/src/object.ts [hunk 1]
```diff
@@ -54,16 +56,31 @@ export function get(
...
+    }
+
+    obj = obj[key[index]]
+  }
+
+  const value = obj === undefined ? fallback : obj
+  map.set(key, value)
+
+  return value
+}
```
Comment (segunadebayo) [inline packages/utils/src/object.ts:83]: Just looking through this and I feel like we can split the cache part and the actual `get` function, then compose them together.

We're doing 2 things within a single function, and I'd like us to separate that.

```js
const memoGet = memoize(get)
```

Can we do this?

Comment (ljosberinn) [reply packages/utils/src/object.ts]: should be resolved, readded memoizedGet where it was removed previously

**Pull request merged**